### PR TITLE
Feat: Add Mint name with crypto modal

### PR DESCRIPTION
**Pull request by LautaroPetaccio**
This PR adds the `MintNameWithCryptoModal` which uses the `BuyWithCryptoModal`.

### CODE REVIEW

File: webapp/src/components/Modals/ClaimNameFatFingerModal/ClaimNameFatFingerModal.container.ts [hunk 1]
```diff
@@ -26,7 +27,10 @@ import {
 import { Contract } from '../../../modules/vendor/services'
 import { getContract } from '../../../modules/contract/selectors'
 import { getMana, getWallet } from '../../../modules/wallet/selectors'
-import { getIsClaimingNamesWithFiatEnabled } from '../../../modules/features/selectors'
+import {
+  getIsClaimingNamesWithFiatEnabled,
+  getIsMintingNamesWithAxelarEnabled
```
Comment (juanmahidalgo) [inline webapp/src/components/Modals/ClaimNameFatFingerModal/ClaimNameFatFingerModal.container.ts:32]: nit-pick: can we change the name `Axelar` for `CrossChain` to keep consistency?

**Review by juanmahidalgo — APPROVED** [review]
LGTM!! great job 👏👏👏

File: webapp/src/modules/ens/actions.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { action } from 'typesafe-actions'
 import { ChainId } from '@dcl/schemas'
 import { buildTransactionPayload } from 'decentraland-dapps/dist/modules/transaction/utils'
 import { ENS, ENSError } from './types'
+import { Route } from 'decentraland-transactions/crossChain'
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply webapp/src/modules/ens/actions.ts]: Moved it!

File: webapp/src/modules/ens/sagas.ts [hunk 3]
```diff
@@ -74,7 +87,12 @@ export function* ensSaga() {
           content: ethers.constants.AddressZero,
           contractAddress: dclRegistrarContract.address
         }
-        yield put(claimNameSuccess(ens, subdomain, hash))
+
+        if (isCrossChain) {
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply webapp/src/modules/ens/sagas.ts:91]: As we split the loading selector into buying natively and cross chain, this action should eventually end up clearing the cross chain request action loader. I'll keep it for now and I'll try to refactor the modal into a single loader to later see if we can simplify this and other requests as well.

File: webapp/src/components/Modals/BuyWithCryptoModal/hooks.ts [hunk 4]
```diff
@@ -246,8 +245,15 @@ export const useBuyNftGasCost = (
     selectedChain,
     order.network
   )
+
+  if (order.price === '0' || isPriceTooLow(order.price)) {
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply webapp/src/components/Modals/BuyWithCryptoModal/hooks.ts]: Removed it!

File: webapp/src/components/Modals/BuyWithCryptoModal/hooks.ts [hunk 5]
```diff
@@ -210,8 +203,14 @@ export const useMintingNftGasCost = (
     selectedChain,
     item.network
   )
+  if (item.price === '0' || isPriceTooLow(item.price)) {
+    return {
+      gasCost: undefined,
+      isFetchingGasCost: false
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply webapp/src/components/Modals/BuyWithCryptoModal/hooks.ts]: Removed it!

File: webapp/src/components/Modals/BuyWithCryptoModal/MintNameWithCryptoModal/MintNameWithCryptoModal.tsx [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    isLoadingAuthorization,
+    getContract,
+    onAuthorizedAction,
+    onClaimName,
+    onClaimNameCrossChain,
+    onClose
+  } = props
+
+  const onBuyNatively = useCallback(() => {
+    console.log('Executing on buy natively')
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply webapp/src/components/Modals/BuyWithCryptoModal/MintNameWithCryptoModal/MintNameWithCryptoModal.tsx]: Removed it!

File: webapp/src/components/Modals/BuyWithCryptoModal/MintNameWithCryptoModal/MintNameWithCryptoModal.container.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+import type {
+  MapDispatchProps,
+  MapStateProps,
+  OwnProps
+} from './MintNameWithCryptoModal.types'
+import { MintNameWithCryptoModal } from './MintNameWithCryptoModal'
+
+const mapState = (state: RootState): MapStateProps => {
+  return {
+    isClaimingName: isLoadingType(getLoading(state), CLAIM_NAME_REQUEST),
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply webapp/src/components/Modals/BuyWithCryptoModal/MintNameWithCryptoModal/MintNameWithCryptoModal.container.ts]: Done! I've used Minting as the modal name is called `Mint`.

File: webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx [hunk 8]
```diff
@@ -360,25 +362,28 @@ export const BuyWithCryptoModal = (props: Props) => {
...
-          data-testid={BUY_WITH_CARD_TEST_ID}
-          disabled={isLoading || isLoadingAuthorization}
-          loading={isLoading || isLoadingAuthorization}
-          onClick={handleBuyWithCard}
-        >
-          <Icon name="credit card outline" />
-          {t(`buy_with_crypto_modal.buy_with_card`)}
-        </ChainButton>
+        {onBuyWithCard && (
+          <ChainButton
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx:367]: It seems that we should definitely change it. I'll apply the change in a different PR with more refactors.

File: webapp/src/components/AssetPage/SaleActionBox/BuyNFTButtons/BuyWithCryptoButton/BuyWithCryptoButton.tsx [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import { t } from 'decentraland-dapps/dist/modules/translation'
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply webapp/src/components/AssetPage/SaleActionBox/BuyNFTButtons/BuyWithCryptoButton/BuyWithCryptoButton.tsx:1]: Removed it!

**Pull request merged**